crates/utils/src/tracing/grpc.rs

```diff
@@ -9,6 +9,30 @@ macro_rules! rpc_span {
     };
 }
 
+/// A [`trace_fn`](tonic::transport::server::Server) implementation for the RPC which
+/// adds open-telemetry information to the span.
+///
+/// Creates an `info` span following the open-telemetry standard: `rpc.rpc/{method}`.
+/// Additionally also pulls in remote tracing context which allows the server trace to be connected
+/// to the client's origin trace.
+pub fn rpc_trace_fn<T>(request: &http::Request<T>) -> tracing::Span {
+    let span = match request.uri().path().rsplit('/').next() {
+        Some("CheckNullifiers") => rpc_span!("rpc.rpc", "CheckNullifiers"),
+        Some("CheckNullifiersByPrefix") => rpc_span!("rpc.rpc", "CheckNullifiersByPrefix"),
+        Some("GetBlockHeaderByNumber") => rpc_span!("rpc.rpc", "GetBlockHeaderByNumber"),
+        Some("SyncState") => rpc_span!("rpc.rpc", "SyncState"),
+        Some("SyncNotes") => rpc_span!("rpc.rpc", "SyncNotes"),
+        Some("GetNotesById") => rpc_span!("rpc.rpc", "GetNotesById"),
+        Some("SubmitProvenTransaction") => rpc_span!("rpc.rpc", "SubmitProvenTransaction"),
+        Some("GetAccountDetails") => rpc_span!("rpc.rpc", "GetAccountDetails"),
+        Some("GetBlockByNumber") => rpc_span!("rpc.rpc", "GetBlockByNumber"),
+        Some("GetAccountStateDelta") => rpc_span!("rpc.rpc", "GetAccountStateDelta"),
+        Some("GetAccountProofs") => rpc_span!("rpc.rpc", "GetAccountProofs"),
+        _ => rpc_span!("rpc.rpc", "Unknown"),
+    };
+    add_otel_span_attributes(span, request)
```

Comment:
Do we want this? This would allow external callers to inject open-telemetry spans I think, iiuc.